### PR TITLE
Added logHeaders and logRequestPayload and logResponsePayload options closes #36

### LINKED ISSUE #36: Good options logRequestHeaders and logRequestPayload not logged to console
**Expected**
When logRequestHeaders and logRequestPayload are set to true, it's expected that good-console will log the headers and payload directly to console.

**Actual**
good-console does not log either request header or post payload

**Example Options**
```
var options = {
    opsInterval: 1000,
    reporters: [{
      reporter: require('good-console'),
      args:[{ log: '*', request: '*', response: '*' }]
    }],
    logRequestHeaders: true,
    logRequestPayload: true,
    logResponsePayload: true
  }
```

### PR DESCRIPTION
**Pull request by AdriVanHoudt**
closes #36 

As you can see on line 199-206 I removed some trailing spaces. I consider this a bug in previous versions since they were there because of of Hoek.format leaving a space there when %s is not filled.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,6 +19,8 @@ Creates a new GoodConsole object with the following arguments:
 - `[options]` - optional object with the following available keys
 	- `format` - [MomentJS](http://momentjs.com/docs/#/displaying/format/) format string. Defaults to 'YYMMDD/HHmmss.SSS'.
 	- `utc` - boolean controlling Moment using [utc mode](http://momentjs.com/docs/#/parsing/utc/) or not. Defaults to `true`.
+	- `logHeaders` - log the headers from the request if present. You have to enable `requestHeaders` in Good for this. Defaults to false.
```
Comment (arb) [inline README.md]: Warp "false" in single ticks since it's technically code.

File: lib/index.js [hunk 2]
```diff
@@ -62,9 +64,9 @@ internals.GoodConsole.prototype.init = function (stream, emitter, callback) {
 
         if (eventName === 'ops') {
             eventPrintData.data = Hoek.format('memory: %sMb, uptime (seconds): %s, load: %s',
-                Math.round(data.proc.mem.rss / (1024 * 1024)),
-                data.proc.uptime,
-                data.os.load);
+                                              Math.round(data.proc.mem.rss / (1024 * 1024)),
```
Comment (arb) [inline lib/index.js]: Weird spacing.

File: lib/index.js [hunk 3]
```diff
@@ -131,7 +127,14 @@ internals.GoodConsole.prototype._formatResponse = function (event, tags) {
     };
     var color = methodColors[event.method] || 34;
     var method = '\x1b[1;' + color + 'm' + event.method + '\x1b[0m';
+    data.push(method);
```
Comment (arb) [inline lib/index.js]: Rather than do two `data.push`s back to back, just push multiple values at once.

File: lib/index.js [hunk 4]
```diff
@@ -143,11 +146,24 @@ internals.GoodConsole.prototype._formatResponse = function (event, tags) {
         }
         statusCode = '\x1b[' + color + 'm' + event.statusCode + '\x1b[0m';
     }
+    data.push(statusCode);
+
+    data.push('(' + event.responseTime + 'ms)');
+
+    if (this._settings.logHeaders && typeof event.headers === 'object' && event.headers) {
```
Comment (arb) [inline lib/index.js]: I don't think you need to check the `typeof` here, should always be an object.

File: lib/index.js [hunk 5]
```diff
@@ -143,11 +146,24 @@ internals.GoodConsole.prototype._formatResponse = function (event, tags) {
...
+        data.push('headers: ' + SafeStringify(event.headers));
+    }
+    if (this._settings.logPayload && typeof event.requestPayload === 'object' && event.requestPayload) {
+        data.push('request payload: ' + SafeStringify(event.requestPayload));
+    }
+    if (typeof event.responsePayload === 'object' && event.responsePayload) {
```
Comment (arb) [inline lib/index.js]: Should be a setting for this too.

File: lib/index.js [hunk 6]
```diff
@@ -143,11 +146,24 @@ internals.GoodConsole.prototype._formatResponse = function (event, tags) {
...
+        data.push('response payload: ' + SafeStringify(event.responsePayload));
+    }
 
     return this._printEvent({
         timestamp: event.timestamp,
         tags: tags,
-        //instance, method, path, query, statusCode, responseTime, responsePayload
-        data: Hoek.format('%s: %s %s %s %s (%sms) %s', event.instance, method, event.path, query, statusCode, event.responseTime, responsePayload)
+        //instance, method, path, query, statusCode, responseTime, requestHeaders, requestPayload, responsePayload
+        data: Hoek.format('%s: %s', event.instance, data.join(' '))
```
Comment (arb) [inline lib/index.js]: Might as well just use `data.join` for everything and skip `Hoek.format` entirely at this point.

File: lib/index.js (hunk 5 above)
Comment (AdriVanHoudt) [inline lib/index.js]: `logRequestPayload` and `logResponsePayload` then? also `logResponsePayload`  should default to true to no be breaking

File: lib/index.js [hunk 7]
```diff
@@ -143,11 +145,24 @@ internals.GoodConsole.prototype._formatResponse = function (event, tags) {
         }
         statusCode = '\x1b[' + color + 'm' + event.statusCode + '\x1b[0m';
     }
+    data.push(statusCode);
```
Comment (arb) [inline lib/index.js]: Push both items in one call.

**Comment by arb** [comment]
After reflecting on this, I don't think I'm going to add this feature. You shouldn't be logging this kind of information to the console. It's too much and is essentially useless. I will reconsider if more people want this, but this has been sitting here for months with no prodding.

**Comment by AdriVanHoudt** [comment]
it's probably more of a consistency thing and in my case it would be helpful for debugging

**Pull request closed without merging**